src/Database/Bloodhound/Types.hs

```diff
@@ -1744,9 +1746,20 @@ class BucketAggregation a where
 
 data Bucket a = Bucket { buckets :: [a]} deriving (Show)
 
-data TermsResult = TermsResult { termKey       :: Text
-                               , termsDocCount :: Int
-                               , termsAggs     :: Maybe AggregationResults } deriving (Show)
+data TermsResult
+  = TextTermsResult
```

Comment:
Ah. Thank god for tests. Rebased to https://github.com/MHova/bloodhound/commit/cba1a574c3693fa7845212393025ca48fcd4890f